plugins/MultiSites/Dashboard.php

```diff
@@ -61,11 +65,14 @@ public function __construct($period, $date, $segment)
                     $pastRow = $pastData->getRowFromLabel($idSite);
                     if ($pastRow) {
                         $pastRow->setColumn('label', $site['name']);
-                        $pastData->setLabelsHaveChanged();
```

Comment:
This here causes a huge performance regression when having eg 30k sites. It made it possibly even slower than before. Meaning we spent lots of time to improve performance just to make it slow again a week later. We need performance regression tests.